Core/Graphics/RHI/METALCommon.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <Metal/Metal.hpp>
+
+struct METALCompileObjects
+{
+    MTL::Library** library;
```

Comment:
Do we have to use raw pointers? Can they be wrapped in some smart pointer. Similar to `ComPtr` for D3D